src/Clash/Cores/Ethernet/Arp.hs

```diff
@@ -0,0 +1,93 @@
+{-# language BlockArguments #-}
+{-# language FlexibleContexts #-}
+{-# language RecordWildCards #-}
+
+{-|
+Module      : Clash.Cores.Ethernet.Arp
+Description : Top level ARP module.
+-}
+module Clash.Cores.Ethernet.Arp
+  ( arpTransmitter
+  , arpReceiverC
+  ) where
+
+import Clash.Prelude
+
+import Protocols
+import Protocols.Df qualified as Df
+import Protocols.Df.Extra ( filterS, partitionS )
+import Protocols.Extra.PacketStream
+import Protocols.Extra.PacketStream.Packetizers ( depacketizeToDfC, packetizeFromDfC )
+
+import Clash.Cores.Ethernet.Arp.ArpTypes
+import Clash.Cores.Ethernet.IP.IPv4Types
+import Clash.Cores.Ethernet.Mac.EthernetTypes
+
+
+-- | Transmits ARP packets upon request.
+arpTransmitter
```

Comment:
The definition need to moved to innside the `Arp` directory. The module which will come here will be implemented by https://github.com/GiPHouse/qbaylogic-clash-based-macipudp-stack-spring24/pull/138

So for now you can delete this file and move the definitions `ArpManager`.

Reply:
The ICMP top level module also contains the transmitter and receiver, should those be moved to something like ICMP manager as well?

Reply:
There will be no ICMP manager.